lib/praxis/response_definition.rb

```diff
@@ -287,10 +287,17 @@ def validate_body!(response)
     def validate_parts!(response)
       return unless parts
 
-      response.parts.each do |name, part|
-        parts.validate(part)
+      case response.body
+      when Praxis::Types::MultipartArray
+        response.body.each do |part|
```

Comment:
shouldn't we call response.body.validate? ... and have that loop through the parts...and possibly allow the type to validate any other necessary thing outside the parts (i.e., like the preface or whatever...)